src/test/shell/bazel/external_patching_test.sh

```diff
@@ -16,13 +16,35 @@
 #
 # Tests the patching functionality of external repositories.
 
-# Load the test setup defined in the parent directory
-CURRENT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
-source "${CURRENT_DIR}/../integration_test_setup.sh" \
+set -euo pipefail
+# --- begin runfiles.bash initialization ---
+if [[ ! -d "${RUNFILES_DIR:-/dev/null}" && ! -f "${RUNFILES_MANIFEST_FILE:-/dev/null}" ]]; then
+  if [[ -f "$0.runfiles_manifest" ]]; then
+    export RUNFILES_MANIFEST_FILE="$0.runfiles_manifest"
+  elif [[ -f "$0.runfiles/MANIFEST" ]]; then
+    export RUNFILES_MANIFEST_FILE="$0.runfiles/MANIFEST"
+  elif [[ -f "$0.runfiles/bazel_tools/tools/bash/runfiles/runfiles.bash" ]]; then
+    export RUNFILES_DIR="$0.runfiles"
+  fi
+fi
+if [[ -f "${RUNFILES_DIR:-/dev/null}/bazel_tools/tools/bash/runfiles/runfiles.bash" ]]; then
+  source "${RUNFILES_DIR}/bazel_tools/tools/bash/runfiles/runfiles.bash"
+elif [[ -f "${RUNFILES_MANIFEST_FILE:-/dev/null}" ]]; then
+  source "$(grep -m1 "^bazel_tools/tools/bash/runfiles/runfiles.bash " \
+            "$RUNFILES_MANIFEST_FILE" | cut -d ' ' -f 2-)"
+else
+  echo >&2 "ERROR: cannot find @bazel_tools//tools/bash/runfiles:runfiles.bash"
+  exit 1
+fi
+# --- end runfiles.bash initialization ---
+
+source "$(rlocation "io_bazel/src/test/shell/integration_test_setup.sh")" \
   || { echo "integration_test_setup.sh not found!" >&2; exit 1; }
-source "${CURRENT_DIR}/remote_helpers.sh" \
-  || { echo "remote_helpers.sh not found!" >&2; exit 1; }
 
+if is_windows; then
```

Comment:
Copy this:
https://github.com/bazelbuild/bazel/blob/4fbeabdb5da1a22e2f0e69c1349176e65f9aa093/src/test/shell/integration/incompatible_changes_conflict_test.sh#L48-L68

(Also in the other tests.)

`is_windows` is a function defined in `test-setup.sh` (and as such will go away with the [new test wrapper](https://github.com/bazelbuild/bazel/issues/5508)) while `$is_windows` is a variable defined in the code I just pasted.

Reply:
Looks like `is_windows` is defined in `testenv.sh` instead of `test-setup.sh`. And `testenv.sh` is only used for shell tests. So I think it's fine to use it here?

https://github.com/bazelbuild/bazel/blob/master/src/test/shell/testenv.sh#L24

Reply:
Ah, OK. Thanks for checking!